src/Microsoft.Extensions.Primitives/StringSegment.cs

```diff
@@ -30,22 +54,61 @@ public StringSegment(string buffer, int offset, int length)
             Length = length;
         }
 
+        /// <summary>
+        /// The <see cref="string"/> that acts as a buffer for this segment.
+        /// </summary>
         public string Buffer { get; }
 
+        /// <summary>
+        /// The offset within the Buffer at wich the segment starts.
+        /// </summary>
         public int Offset { get; }
 
+        /// <summary>
+        /// The length of the segment starting from the offset defined by the segment.
+        /// </summary>
         public int Length { get; }
```

Comment:
You can clean up all of the docs for these properties to be less verbose as well. It's not saying anything really important.